migrator/cmd.go

```diff
@@ -0,0 +1,230 @@
+				err := m.k8sClient.BatchV1().Jobs(runCommandNamespace).Delete(job.Name, &apismetav1.DeleteOptions{})
+
+				fmt.Printf("Job %s was completed.\n", job.Name)
+				if err != nil {
+					return microerror.Mask(err)
+				}
+				break
+			}
+		}
+	}
+
+	return nil
+}
+
+func isJobCompleted(j *batchapiv1.Job) bool {
+	for _, c := range j.Status.Conditions {
+		if c.Type == "Complete" && c.Status == apiv1.ConditionTrue {
+			return true
+		}
+	}
+	return false
+}
+
+// buildCommandJob return job that will execute commands on a node in host namespace.
+// The executed file is taken from configmap which is mounted to the pod.
+func buildCommandJob(nodeName string, dockerRegistry string) *batchapiv1.Job {
+	activeDeadlineSeconds := int64(120)
+	backOffLimit := int32(10)
+	completions := int32(1)
+	privileged := true
+	priority := int32(2000000000)
+	parallelism := int32(1)
+	jobName := fmt.Sprintf("%s-command", project.Name())
+	cpu := resource.MustParse("50m")
+	memory := resource.MustParse("50Mi")
+
+	j := batchapiv1.Job{
+		TypeMeta: apismetav1.TypeMeta{
+			Kind:       "Job",
+			APIVersion: batchapiv1.GroupName,
+		},
+		ObjectMeta: apismetav1.ObjectMeta{
+			Name:      jobName,
+			Namespace: runCommandNamespace,
+			Labels: map[string]string{
+				"app":        jobName,
+				"created-by": project.Name(),
+			},
+		},
+		Spec: batchapiv1.JobSpec{
+			Parallelism:  &parallelism,
+			Completions:  &completions,
+			BackoffLimit: &backOffLimit,
+			Template: apiv1.PodTemplateSpec{
+				Spec: apiv1.PodSpec{
+					Containers: []apiv1.Container{
+						{
+							Name:  "run-command",
+							Image: jobDockerImage(dockerRegistry),
+							Resources: apiv1.ResourceRequirements{
+								Limits: apiv1.ResourceList{
+									apiv1.ResourceCPU:    cpu,
+									apiv1.ResourceMemory: memory,
+								},
+								Requests: apiv1.ResourceList{
+									apiv1.ResourceCPU:    cpu,
+									apiv1.ResourceMemory: memory,
+								},
+							},
+							SecurityContext: &apiv1.SecurityContext{
+								Privileged:               &privileged,
+								AllowPrivilegeEscalation: &privileged,
+							},
+							Command: []string{
+								"/bin/sh",
+								"/data/command.sh",
+							},
+							VolumeMounts: []apiv1.VolumeMount{
+								{
+									Name:      runCommandVolume,
+									MountPath: "/data/",
+									ReadOnly:  true,
+								},
+							},
+						},
+					},
+					HostPID: true,
+					NodeSelector: map[string]string{
+						"kubernetes.io/hostname": nodeName,
+					},
+					RestartPolicy:      apiv1.RestartPolicyNever,
+					Priority:           &priority,
+					PriorityClassName:  runCommandPriorityClass,
+					ServiceAccountName: runCommandSAName,
+					Tolerations: []apiv1.Toleration{
+						{
+							Key:      "node.kubernetes.io/unschedulable",
+							Operator: "Exists",
+							Effect:   "NoSchedule",
+						},
+						{
+							Key:    "node-role.kubernetes.io/master",
+							Effect: "NoSchedule",
+						},
+					},
```

Comment:
FMI, I can never really read and understand what the tolerations mean. Does that mean that the job is tolerating the "master no schedule" taint or that it cannot be scheduled on nodes defining this toleration? 🙈

Reply:
it basically means, ignores this taint 

so schedule even if this taint is on that node